### DISCUSSION TITLE
Welcome to Jarvis_AI Discussions!

### DISCUSSION THREAD

**Discussion by Dipeshpal** [Dipeshpal/Jarvis_AI#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by siddharth403** [comment]
Reminder Function by creating a variable when the user asks

**Reply by Dipeshpal** [reply]
**WIP**

**Comment by siddharth403** [comment]
News calling from this video https://www.bbc.co.uk/news/av/10462520
Perhaps downloading the audio file and playing the audio file.

**Reply by Dipeshpal** [reply]
copyright or legal issues, to implement this.

**Comment by siddharth403** [comment]
Timer function

"Set a timer" -> "Okay, for how long?" ->
 "5 minutes" -> "Sure, 5 minutes, starting now!"

**Comment by siddharth403** [comment]
Also features from the movie
Like 'there terabytes of calculations needed before an actual flight can be taken.'

**Reply by Dipeshpal** [reply]
sorry, I didn't understand. Please elaborate.

**Reply by Akul2010** [reply]
Do you mean like features from actual JARVIS from Iron Man?

**Comment by navintech93** [comment]
Hi, does this has multi-language support, like speech in Indian languages . If not i am interested to add those feature

**Reply by Dipeshpal** [reply]
yes, use-
res = obj.mic_input(lang='en-IN')

NOW, you might need to convert "res" (Hindi) text back to English. Then feed English text to other functions.

Example-
```
res = obj.mic_input(lang='en-IN')
res_eng = some_lang_translator(res) # it should give response in English
                if re.search("jokes|joke|Jokes|Joke", res_eng):
                    joke_ = obj.tell_me_joke('en', 'neutral')
                    print(joke_)
                    t2s(joke_)


```

check the below link for supported language.

https://cloud.google.com/speech-to-text/docs/languages

NOTE-

mic_input_ai()  **#doesn't support multiple language**

**Comment by vigneshvikey919** [comment]
How to install pip

**Reply by Dipeshpal** [reply]
google it

**Comment by iamDyeus** [comment]
Which tts Module is it Using? 
Is it the basic pyttsx3 module or the google Cloud speech or some kind of API ?

**Reply by Dipeshpal** [reply]
import speech_recognition as sr
import os
from gtts import gTTS
from playsound import playsound

Check: https://github.com/Dipeshpal/Jarvis_AI/blob/07b003f61d8772591aaffbb6ea76a49bffeecbcf/JarvisAI/JarvisAI/JarvisAI/__init__.py#L191

**Comment by anshaggarwal28** [comment]
line 11, in <module>
    obj = JarvisAI.JarvisAssistant(sync=True, token='5ec64be7ff718ac25917c198f3d7a4', disable_msg=False,
TypeError: JarvisAssistant.__init__() got an unexpected keyword argument 'sync'

error

**Reply by Dipeshpal** [reply]
read docs

**Comment by ingjrpp** [comment]
Hello, I congratulate you on the project, I would like to know if there is a way to change the language? for example to speak to you and respond in Spanish?

**Comment by vednig** [comment]
**Feature/Improvement:**
- Provide .exe file for direct download

**Reply by Jaskawal-Singh** [reply]
yes

**Comment by Jaskawal-Singh** [comment]
how can i open that whole folder in the pyscripter app ??

**Comment by afflictyou** [comment]
Does JarvisAI 4.0 require a PyArmor license to use it?  I'm getting a "Check license failed, Invalid input packet."

**Comment by RSlient007** [comment]
I suggest we need a discord server to communicate :))))
So what do you think?
